status/reporters.go

```diff
@@ -625,13 +625,24 @@ func (r *GitHubReporter) ReportStatusForSnapshot(k8sClient client.Client, ctx co
 
 		for _, integrationTestStatusDetail := range integrationTestStatusDetails {
 			integrationTestStatusDetail := *integrationTestStatusDetail // G601
+
+			pipelineRun, err := helpers.GetPipelineWithName(integrationTestStatusDetail.TestPipelineRunName, snapshot.Namespace, k8sClient, ctx)
+			if err != nil {
+				return err
+			}
+
 			checkRun, err := r.createCheckRunAdapterForSnapshot(snapshot, integrationTestStatusDetail, owner, repo, sha)
 			if err != nil {
 				logger.Error(err, "failed to create checkRunAdapter for snapshot",
 					"snapshot.NameSpace", snapshot.Namespace, "snapshot.Name", snapshot.Name)
 				return err
 			}
 
+			err = helpers.RemoveFinalizer(k8sClient, *logger, ctx, pipelineRun)
```

Comment:
How do we know that we can already remove finalizers here, if I'm not mistaken, this function also reports other statuses as well, so you mey remove finalizer when PLR is still in progress.

Reply:
Thanks for the info, this logic now gets called after all the status reporting is completed.